src/formats/onenote.ts

```diff
@@ -0,0 +1,489 @@
+
+			let label = listElement.createEl('label');
+
+			let checkbox = label.createEl('input');
+			checkbox.type = 'checkbox';
+
+			label.appendChild(document.createTextNode(section.displayName!));
+			label.createEl('br');
+
+			// Add/remove a section from this.selectedSections
+			checkbox.addEventListener('change', () => {
+				if (checkbox.checked) this.selectedSections.push(section);
+				else {
+					const index = this.selectedSections.findIndex((sec) => sec.id === section.id);
+					if (index !== -1) {
+						this.selectedSections.splice(index, 1);
+					}
+				}
+			});
+		});
+	}
+
+	async import(progress: ProgressReporter): Promise<void> {
+		let outputFolder = await this.getOutputFolder();
+		if (!outputFolder) {
+			new Notice('Please select a location to export to.');
+			return;
+		}
+
+		for (let section of this.selectedSections) {
+			let sectionFolder: TFolder = await this.createFolders(outputFolder.path + '/' + section.displayName);
+
+			let pages: OnenotePage[] = (await (await fetch(`
+					https://graph.microsoft.com/v1.0/me/onenote/sections/${section.id}/pages?$select=id,title,createdDateTime,lastModifiedDateTime`,
+			this.requestParams)).json()).value;
+
+			pages.forEach(async (page) =>
+				this.processFile(progress, sectionFolder, await (await fetch(
+					`https://graph.microsoft.com/v1.0/me/onenote/pages/${page.id}/content?includeInkML=true`,
+					this.requestParams)).text(), page));
+		}
+	}
+
+	async processFile(progress: ProgressReporter, folder: TFolder, content: string, page: OnenotePage) {
+		try {
+			const splitContent = this.convertFormat(content);
+
+			let parsedPage: HTMLElement = this.getAllAttachments(splitContent.html);
+			parsedPage = this.styledElementToHTML(parsedPage);
+			parsedPage = this.fixTables(parsedPage);
+			parsedPage = this.convertTags(parsedPage);
+			parsedPage = this.convertInternalLinks(parsedPage);
+			parsedPage = await this.convertDrawings(parsedPage);
+
+			let mdContent = htmlToMarkdown(parsedPage).trim();
+			const fileRef = await this.saveAsMarkdownFile(folder, page.title!, mdContent);
+
+			await this.fetchAttachmentQueue(progress, fileRef);
+
+			// Add the last modified and creation time metadata
+			const writeOptions: DataWriteOptions = {
+				ctime: Date.parse(page.createdDateTime!.toString()) ||
+					   Date.parse(page.lastModifiedDateTime!.toString()) ||
+					   Date.now(),
+				mtime: Date.parse(page?.lastModifiedDateTime!.toString()) ||
+					   Date.parse(page?.createdDateTime!.toString()) ||
+					   Date.now(),
+			};
+			await this.vault.append(fileRef, '', writeOptions);
+			progress.reportNoteSuccess(page.title!);
+		}
+		catch (e) {
+			progress.reportFailed(page.title!, e);
+		}
+	}
+
+	// OneNote returns page data and inking data in one file, so we need to split them
+	convertFormat(input: string): { html: string; inkml: string } {
+		const output = { html: '', inkml: '' };
+		const boundary = input.split('\n')[0];
```

Comment:
If we're only using the first line, then we can improve performance by only splitting out to the first line break.

```suggestion
		const boundary = input.split('\n', 1)[0];
```